Makefile

```
@@ -20,6 +20,10 @@ test-all: lint test
 test:
 	go test -race ./...
 
+.PHONY: test-with-coverage
+test-with-coverage:
+	go test -race -v -coverprofile=coverage.out ./...
```

```suggestion
	go test -race -v -count=1 -coverprofile=coverage.out ./...
```

For consistency with `.PHONY`. If the intention is that this target re-runs the whole battery of tests, making it `.PHONY` won't suffice. Enforcing `-count=1` is necessary to prevent Go toolchain from caching test results.

On another note: I'm not sure how test result caching affects coverage. :thinking:

---

the caching shouldn't cause any problems and I'm tempted to suggest that we in fact allow the use of the cache to build coverage, as this is generally ideal? Let me know your thoughts.

---

I don't see a good case for making the target `.PHONY` but not disabling caching.

If our tests happen to become flaky - re-running this (`.PHONY` but not `-count=1`) will potentially hide that flaw.

Considering three options:
1. `.PHONY` and `-cover=1`
2. not `.PHONY` and not `-cover=1`
3. `.PHONY` but not `-cover=1`

only 1 ("always re-run everything") and 2 ("re-generate the coverage file only for code files that have changed") make sense to me. I do not understand why 3 would be better than the other two. What do you think?

---

makes sense to me :+1:

---

I've opted for the second option to keep the caching, let me know if you want any other changes in this regard :vulcan_salute:

---

FWIW, this currently doesn't work as intended because this make rule gets re-executed all the time unless we provide a list of prerequisites in this rule (e.g. using `go list ./...` - several options exist here, choosing and implementing one of them for all make rules is the clue of #934).

In other words, fixing the make rule dependency problem we have (=#934) appears like a prerequisite for (2) to me.

How would you like to proceed here? I don't have a strong preference:

- (1) would be my choice, but up to you
- (2) requires fixing #934
- (3) is somewhat inconsistent, but doing (3) and waiting for #934 to turn it into (2) isn't _that_ bad :)

---

Given the extra context I've decided to go with your choice in the matter :+1: